pkg/providers/ovirt/provider.go

```diff
@@ -169,6 +177,16 @@ func (o *OvirtProvider) StopVM() error {
 	return nil
 }
 
+// FindTemplate attempts to find best match for a template based on the source VM
+func (o *OvirtProvider) FindTemplate() (*templatev1.Template, error) {
```

Comment:
Will templates be used only for oVirt? Does it have to be in the provider API? It looks to be provider-agnostic and potentially movable to the controller.

Reply:
It shouldn't be oVirt specific and should be provider agnostic. ATM is part of ovirt provider be we can move it outside in different patch.